WalletWasabi.Fluent/Models/Wallets/Address.cs

```diff
@@ -0,0 +1,79 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin;
+using ReactiveUI;
+using WalletWasabi.Blockchain.Analysis.Clustering;
+using WalletWasabi.Blockchain.Keys;
+using WalletWasabi.Extensions;
+using WalletWasabi.Hwi;
+using WalletWasabi.Logging;
+using WalletWasabi.Wallets;
+
+namespace WalletWasabi.Fluent.Models.Wallets;
+
+public class Address : ReactiveObject, IAddress
+{
+	public Address(Wallet wallet, HdPubKey hdPubKey)
+	{
+		Wallet = wallet;
```

Comment:
`Wallet` is not necessary here. What you seems to need is a `KeyManager`

Reply:
@lontivero You're right.
@ichthus1604 the contructor should then turn into this:

```
	public Address(Network network, KeyManager keyManager, HdPubKey hdPubKey)
	{
		HdPubKey = hdPubKey;
		Network = network;
		KeyManager = keyManager;
		HdFingerprint = KeyManager.MasterFingerprint;
		BitcoinAddress = HdPubKey.GetAddress(network);
	}
```

Reply:
`KeyManager` doesn't know about Network?

Reply:
This is maybe what we are looking for.

at KeyManager.cs:
```
	public Network GetNetwork()
	{
		return BlockchainState.Network;
	}
```

Can somebody confirm?

Reply:
yes, it knows the network. You don't need to pass it in the ctor